### PR TITLE
Provide WS-Trust as a connector and re implement WS-Federation using CXF

### PR DESCRIPTION
**Pull request by deshankoswatte**
**Description:**

- Migrate the STS implementation from carbon-identity-framework/security-mgt module and identity-inbound-auth-openid.
- Provide the WS-Trust feature as a connector.
- Re implement WS-Federation using CXF.

**Two new modules:**

- org.wso2.carbon.security.sts.common - Migrated implementation from carbon-identity-framework and identity-inbound-auth-openid repositories.
- org.wso2.carbon.sts.connector - Creates a zip file which provides all the jars needed for the WS-Trust connector.

Depends on https://github.com/wso2/carbon-identity-framework/pull/2921, https://github.com/wso2-extensions/identity-inbound-auth-openid/pull/68 and https://github.com/wso2/wso2-wss4j/pull/82.

Resolves: https://github.com/wso2/product-is/issues/8350

### CODE REVIEW

File: components/org.wso2.carbon.identity.sts.mgt/src/main/java/org/wso2/carbon/identity/sts/mgt/base/IdentityBaseUtil.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
+/*
+ *  Copyright (c) 2005-2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.sts.mgt/src/main/java/org/wso2/carbon/identity/sts/mgt/base/IdentityBaseUtil.java:2]: Lets change the license header.

File: components/org.wso2.carbon.sts.connector/pom.xml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        <groupId>org.wso2.carbon.identity.inbound.auth.sts</groupId>
+        <version>5.6.0-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>org.wso2.carbon.sts.connector</artifactId>
+    <packaging>jar</packaging>
+    <name>WSO2 Carbon - STS Components</name>
+    <description>A custom wso2 products or solution</description>
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.sts.connector/pom.xml:33]: Lets change the name

File: components/org.wso2.carbon.identity.sts.mgt/src/main/java/org/wso2/carbon/identity/sts/mgt/base/IdentityBaseUtil.java (hunk 1 above)
Comment (deshankoswatte) [reply components/org.wso2.carbon.identity.sts.mgt/src/main/java/org/wso2/carbon/identity/sts/mgt/base/IdentityBaseUtil.java]: Fixed with https://github.com/wso2-extensions/identity-inbound-auth-sts/pull/94/commits/5504ab5c7e661e2e7467808b0adfdba1cb3c154a.

File: components/org.wso2.carbon.sts.connector/pom.xml (hunk 2 above)
Comment (deshankoswatte) [reply components/org.wso2.carbon.sts.connector/pom.xml:34]: Any suggestions @madurangasiriwardena?

Comment (deshankoswatte) [reply components/org.wso2.carbon.sts.connector/pom.xml]: Fixed with https://github.com/wso2-extensions/identity-inbound-auth-sts/pull/94/commits/4890afecf4c23fdc6728b191be3025b362cd6576.

**Pull request merged**